pages/api/form/save.ts

```diff
@@ -0,0 +1,45 @@
+import { NextApiRequest, NextApiResponse } from 'next'
+import { WithId, Document, RegExpOrString } from 'mongodb'
+import { FormAnswer, FormValues } from '../../../utils/FormContext'
+import { dbConnect } from '../../../server/_dbConnect'
+import { Question } from '../../../models'
+
+export interface FormAnswerDB extends WithId<Document> {
+  userID: string
+  formValues: FormValues
+  questionHistory: Question[]
+  currentIndex: number
+  currentQuestion: Question
+  currentAnswer: FormAnswer
+}
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  if (req.method !== 'POST') {
+    res.status(400).json({ error: 'Expected POST request' })
+    return
+  }
+
```

Comment:
what are const doc and const client returning?

Reply:
`doc` is short for Document, which is what's stored in the database. So, we want to parse the request body into the form that our database expects it in (to make sure we're not including any additional unnecessary props)

Reply:
`client` is the `MongoClient` which allows us to talk to the databases